### PR TITLE
Allow null values for traits

### PR DESCRIPTION
**Pull request by matthewelwell**

### CODE REVIEW

File: FlagsmithClient.cs [hunk 1]
```diff
@@ -237,9 +237,9 @@ public async Task<Trait> SetTrait(string identity, string key, object value)
         {
             try
             {
-                if (!(value is bool) && !(value is int) && !(value is string))
+                if (!(value is bool) && !(value is int) && !(value is string) && value != null)
                 {
-                    throw new ArgumentException("Value parameter must be string, int or boolean");
+                    throw new ArgumentException("Value parameter must be string, int, boolean or null");
```
Comment (gagantrivedi) [inline FlagsmithClient.cs:242]: I think traits supports `float`  as well?

Comment (matthewelwell) [reply FlagsmithClient.cs]: Yep, good shout. Have added.

**Review by gagantrivedi — DISMISSED** [review]

**Comment by matthewelwell** [comment]
Note that this is waiting on the release of: https://github.com/Flagsmith/flagsmith/pull/916.

File: FlagsmithClient.cs [hunk 2]
```diff
@@ -261,6 +262,19 @@ public async Task<Trait> SetTrait(string identity, string key, object value)
...
+        public async Task SetTraits(string identity, Dictionary<string, object> traits)
+        {
+            ArrayList traitDataObjects = new ArrayList();
+
+            foreach (KeyValuePair<string, object> trait in traits)
+            {
+                traitDataObjects.Add(new { identity = new { identifier = identity }, trait_key = trait.Key, trait_value = trait.Value });
+            }
+
+            string url = configuration.ApiUrl.AppendPath("traits/bulk");
```
Comment (gagantrivedi) [inline FlagsmithClient.cs:274]: Maybe we should add a note that this endpoint is not part of edge and will not exist in v2

**Review by gagantrivedi — APPROVED** [review]

**Pull request merged**